diff --git a/node_modules/patch-package/dist/getPackageResolution.js b/node_modules/patch-package/dist/getPackageResolution.js
index 705fc3b..fecdbe5 100644
--- a/node_modules/patch-package/dist/getPackageResolution.js
+++ b/node_modules/patch-package/dist/getPackageResolution.js
@@ -77,6 +77,9 @@ function getPackageResolution({ packageDetails, packageManager, appPath, }) {
         const lockfile = require(path_1.join(appPath, packageManager === "npm-shrinkwrap"
             ? "npm-shrinkwrap.json"
             : "package-lock.json"));
+        if (lockfile.lockfileVersion > 2) {
+                return Object.entries(lockfile.packages).find(el => el[0].includes(packageDetails.name))[1].resolved;
+            }
         const lockFileStack = [lockfile];
         for (const name of packageDetails.packageNames.slice(0, -1)) {
             const child = lockFileStack[0].dependencies;
